src/WebJobs.Script.WebHost/Extensions/HttpResponseExtensions.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Http
+{
+    public static class HttpResponseExtensions
+    {
+        public static bool HasStarted(this HttpResponse response)
+        {
+            // in some cases HttpResponse.HasStarted isn't returning the correct
+            // value. This extension is a workaround.
+            if (response.HasStarted || response.ContentLength > 0 || response.Body.Length > 0)
```

Comment:
TBD to find out if this is an ASP.NET bug

Reply:
Even before the status code is set for the response, it has a default value of 200, so can't really tell if it has been set

Reply:
That will be problematic in those cases... also, is body always guaranteed to not be `null`?